### PR TITLE
DGL install fixes, dependency updates 

### LINKED ISSUE #326: 🚀[FEA]: Update the DGL build in Dockerfile
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

Medium

### Please provide a clear description of problem you would like to solve.

The source DGL build from Dockerfile no longer works for base images with CUDA > 12.2. 

https://github.com/NVIDIA/modulus/blob/main/Dockerfile#L81

### Describe any alternatives you have considered

The source build was added before DGL had officially shipped a CUDA 12 supported wheel. Now a CUDA 12.1 supported wheel is available on their official website and should be tested to see if the stage can be updated in the Dockerfile.

https://www.dgl.ai/pages/start.html

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
Updates the DGL build in the dockerfile and also updates the default base image. Closes #326 

Also moves `onnx` to the required dependency list as it is required by the inference pytests. 

Changes are tested by running pytests in both deploy and ci containers. 

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG BASE_CONTAINER=nvcr.io/nvidia/pytorch:23.10-py3
+ARG BASE_CONTAINER=nvcr.io/nvidia/pytorch:23.12-py3
```
Comment (NickGeneva) [inline Dockerfile:17]: Wasnt this updated for the release?

File: Dockerfile [hunk 2]
```diff
@@ -67,25 +67,14 @@ RUN if [ "$TARGETPLATFORM" = "linux/arm64" ] && [ -e "/modulus/deps/vtk-9.2.6.de
...
-	git clone https://github.com/dmlc/dgl.git && cd dgl/ && git checkout tags/1.1.2 && git submodule update --init --recursive && \
-	DGL_HOME="/workspace/dgl" bash script/build_dgl.sh -g && \
-	cd python && \
-	python setup.py install && \
-	python setup.py build_ext --inplace; \
-    fi
-RUN rm -rf /workspace/dgl
+
+RUN pip install  dgl -f https://data.dgl.ai/wheels/cu121/repo.html
+RUN pip install  dglgo -f https://data.dgl.ai/wheels-test/repo.html
```
Comment (NickGeneva) [inline Dockerfile:77]: Nice clean up here :)

File: pyproject.toml [hunk 3]
```diff
@@ -24,7 +24,8 @@ dependencies = [
     "pytz>=2023.3",
     "treelib>=1.2.5",
     "tqdm>=4.60.0",
-    "nvtx>=0.2.8"
+    "nvtx>=0.2.8",
+    "onnx>=1.14.0",
```
Comment (NickGeneva) [inline pyproject.toml:28]: Whats onnx added for here?

Comment (NickGeneva) [reply pyproject.toml:28]: Is it DGL requires it?

**Review by NickGeneva — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (ktangsali) [reply Dockerfile:17]: For this release, the base image was changed manually in the internal build pipelines using the `BASE_CONTAINER` arg.

File: pyproject.toml (hunk 3 above)
Comment (ktangsali) [reply pyproject.toml:28]: No, actually we need it for the inference pytests : https://github.com/NVIDIA/modulus/blob/main/test/models/common/inference.py#L19

File: Dockerfile (hunk 2 above)
Comment (ktangsali) [reply Dockerfile:77]: Thanks!

File: Dockerfile [hunk 4]
```diff
@@ -67,25 +67,14 @@ RUN if [ "$TARGETPLATFORM" = "linux/arm64" ] && [ -e "/modulus/deps/vtk-9.2.6.de
...
-        echo "No DGL wheel present, building from source" && \
-	git clone https://github.com/dmlc/dgl.git && cd dgl/ && git checkout tags/1.1.2 && git submodule update --init --recursive && \
-	DGL_HOME="/workspace/dgl" bash script/build_dgl.sh -g && \
-	cd python && \
-	python setup.py install && \
-	python setup.py build_ext --inplace; \
-    fi
-RUN rm -rf /workspace/dgl
+
+RUN pip install  dgl -f https://data.dgl.ai/wheels/cu121/repo.html
```
Comment (Alexey-Kamenev) [inline Dockerfile:76]: 2 things:
1. I would keep `--no-cache-dir` option to keep Docker image smaller (unless DGL will be re-installed later again).
2. There seems to be 2 spaces in `install  dgl`

**Review by Alexey-Kamenev — APPROVED** [review]

File: Dockerfile (hunk 4 above)
Comment (ktangsali) [reply Dockerfile]: Good catch on the `--no-cache-dir` option. I added it for other packages, but somehow forgot for this one. Also fixed the spaces.

**Pull request merged**